src/ui/TreeView.js

```diff
@@ -50,11 +52,14 @@ class TreeView {
     return yo`<ul style=${this.cssList}>${children}</ul>`
   }
 
-  formatDataInternal (key, data, children, expand) {
+  formatDataInternal (key, data, children, properties, expand) {
+    var renderedProperties = Object.keys(properties).map((item) => {
```

Comment:
Also note that the caret will not be visible if the item has properties but no children.